### PR TITLE
Cylera connector

### PR DESCRIPTION
**Pull request by Snick555**
[CON-2956](https://kennasecurity.atlassian.net/browse/CON-2956)
[CON-2960](https://kennasecurity.atlassian.net/browse/CON-2960)
[CON-2961](https://kennasecurity.atlassian.net/browse/CON-2961)
[CON-2962](https://kennasecurity.atlassian.net/browse/CON-2962)
[CON-2963](https://kennasecurity.atlassian.net/browse/CON-2963)
[CON-2982](https://kennasecurity.atlassian.net/browse/CON-2982)
[CON-2983](https://kennasecurity.atlassian.net/browse/CON-2983)

[CON-2956]: https://kennasecurity.atlassian.net/browse/CON-2956?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CON-2960]: https://kennasecurity.atlassian.net/browse/CON-2960?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CON-2961]: https://kennasecurity.atlassian.net/browse/CON-2961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CON-2962]: https://kennasecurity.atlassian.net/browse/CON-2962?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CON-2982]: https://kennasecurity.atlassian.net/browse/CON-2982?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CON-2983]: https://kennasecurity.atlassian.net/browse/CON-2983?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CON-2963]: https://kennasecurity.atlassian.net/browse/CON-2963?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tasks/connectors/cylera/readme.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+| Kenna Asset | from Cylera Vulnerability | Conditions |
+| --- | --- | --- |
+| ip_address | vulnerability.ip_address | |
+| mac_address | vulnerability.mac_address | |
+| tags | ["Vendor:#{vulnerability.vendor}", "Type:#{vulnerability.type}", "Model:#{vulnerability.model}", "Class:#{vulnerability.class}"] | ? |
+
+| Kenna Vulnerability | from Cylera Vulnerability | Conditions |
+| --- | --- | --- |
+| scanner_identifier | vulnerability.vulnerability_name | |
```
Comment (kschultz84) [inline tasks/connectors/cylera/readme.md:51]: Just using vulnerability name seem generic. Is there a unique ID for each vulnerability in the data from cylera?

File: tasks/connectors/cylera/readme.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+| last_seen_at | vulnerability.last_seen | |
+| status | vulnerability.status | |
+| vuln_def_name | vulnerability.vulnerability_name | |
+
+| Kenna Definition | from Cylera Vulnerability and Mitigations | Conditions |
+| --- | --- | --- |
+| scanner_type | "Cylera" | |
+| cve_identifiers | vulnerability.vulnerability_name | if vulnerability.vulnerability_name starts with "CVE" |
+| name | vulnerability.vulnerability_name | |
+| solution | mitigations | |
```
Comment (kschultz84) [inline tasks/connectors/cylera/readme.md:64]: Is there no description returned in the cylera data?

Comment (Snick555) [reply tasks/connectors/cylera/readme.md:64]: No, description is missed there

File: tasks/connectors/cylera/readme.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+| kenna_api_host | false | Kenna API Hostname if not US shared | api.kennasecurity.com |
+| kenna_connector_id | false | If set, we'll try to upload to this connector | n/a |
+| output_directory | false | If set, will write a file upon completion. Path is relative to #{$basedir} | output/cylera |
+
+## Data Mappings
+
+| Kenna Asset | from Cylera Vulnerability | Conditions |
+| --- | --- | --- |
+| ip_address | vulnerability.ip_address | |
+| mac_address | vulnerability.mac_address | |
```
[the remark this answers is not in the document]
Comment (Snick555) [reply tasks/connectors/cylera/readme.md:46]:
```
{
      "ip_address": "10.13.55.144",
      "mac_address": "00:14:38:87:0f:ee",
      "model": "HP LaserJet 1320 Series Printer",
      "type": "Printer",
      "vendor": "HP",
      "class": "Misc IoT",
      "vulnerability_name": "CVE-2006-024",
      "vulnerability_category": "Security",
      "first_seen": 1570527348,
      "last_seen": 1576303348,
      "severity": "Medium",
      "status": "Open",
      "confidence": "High"
    }
    ```
    Here is a returned json from Cylera so there no other locators except IP and mac addresses

File: tasks/connectors/cylera/readme.md (hunk 1 above)
Comment (Snick555) [reply tasks/connectors/cylera/readme.md:51]: No other uniq fields in the response from Cylera which we can use for it

**Review by jgarber-cisco — APPROVED** [review]
Nice work!

**Pull request merged**